lib/percy.rb

```diff
@@ -0,0 +1,119 @@
+require 'logger'
+require 'net/http'
+require 'uri'
+require 'json'
+require 'environment'
+
+module Percy
+  # Takes a snapshot of the given page HTML and its assets.
+  #
+  # See https://docs.percy.io/v1/docs/configuration for detailed documentation on
+  # snapshot options.
+  #
+  # @param [Capybara::Session] page The Capybara page to snapshot.
+  # @param [Hash] options
+  # @option options [String] :name A unique name for the current page that identifies
+  #   it across builds. By default this is the URL of the page, but can be customized if the
+  #   URL does not entirely identify the current state.
+  # @option options [Array(Number)] :widths Widths, in pixels, that you'd like to capture for
+  #   this snapshot.
+  def self.snapshot(page, options = {})
+    return unless self._is_agent_running?
+
+    domSnapshot = self._make_dom_snapshot(page)
+    return unless domSnapshot
+
+    if !options.has_key?(:name)
+      options[:name] = page.current_url
+    end
+
+    body = {
+      url: page.current_url,
+      domSnapshot: domSnapshot,
+      clientInfo: Percy.client_info,
+      environmentInfo: Percy.environment_info,
+    }
+    body = body.merge(self._keys_to_json(options))
+    self._post_snapshot_to_agent(body)
+  end
+
+  private
+
+  AGENT_HOST = 'localhost'
+  # Technically, the port is configurable when you run the agent. One day we might want
+  # to make the port configurable in this SDK as well.
+  AGENT_PORT = 5338
+  AGENT_JS_PATH= '/percy-agent.js'
+
+  def self._logger
+    unless defined?(@logger)
+      @logger = Logger.new(STDOUT)
+      @logger.formatter = proc do |_severity, _datetime, _progname, msg|
+        "[percy] #{msg} \n"
+      end
+    end
+    return @logger
+  end
+
+  def self._get_agent_js
+    begin
+      return Net::HTTP.get(AGENT_HOST, AGENT_JS_PATH, AGENT_PORT)
+    rescue => e
+      self._logger.error { "Could not load #{AGENT_JS_PATH}. Error: #{e}"}
+      return nil
+    end
+  end
+
+  def self._make_dom_snapshot(page)
+    agent_js = self._get_agent_js
+    return unless agent_js
+
+    begin
+      page.execute_script(agent_js)
+      dom_snapshot_js = <<-JS
+      (function() {
+        var percyAgentClient = new PercyAgent({ handleAgentCommunication: false });
+        return percyAgentClient.snapshot('unused');
+      })()
+      JS
+      return page.evaluate_script(dom_snapshot_js)
+    rescue => e
+      self._logger.error { "Snapshotting failed. Note that Poltergeist and Rake::Test are no longer supported by percy-capybara. Error: #{e}"}
```

Comment:
@anaulin This line should be throwing a lint error — just noticed that we don't run `bundle exec rubocop -D` in the circle steps like we used to in the travis yml.

Reply:
We do run `bundle exec rubocop -D` in CI in this project. The line in the config is here: https://github.com/percy/percy-capybara/blob/master/.circleci/config.yml#L17

And you can see it running and passing in CI, on this PR here: https://circleci.com/gh/percy/percy-capybara/154?utm_campaign=vcs-integration-link&utm_medium=referral&utm_source=github-build-link

So that line passes the linter, and I'm not sure what would you like to be changed in this line. :-)

Reply:
Added the missing space before the `}`, in case that's what you were pointing at.